src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -629,6 +682,12 @@ else if (!"MQTT".equals(mqttConnect.protocolName().asString()) || mqttConnect.pr
         return progress;
     }
 
+    private boolean checkWillFlag(
```

Comment:
Suggest renaming to `isSetWillFlag` and making this `static` at the end of the factory class.
Suggest adding other similar methods to manage flags and make the conditional logic above more readable.